rmw_zenoh_cpp/src/rmw_subscriber.cpp

```diff
@@ -113,6 +114,9 @@ rmw_create_subscription(
   subscription->options = *subscription_options;
   subscription->can_loan_messages = false;
 
+  // NOTE(CH3): Does this leak because it can't be freed?
```

Comment:
Ok, will do for all the other duplicated strings

Reply:
I've opened an issue to track this. https://github.com/osrf/rmw_zenoh/issues/20

Though I have the changes saved locally on my machine, I think it would be neater to resolve this in a separate PR.

Reply:
Updated: 23497d4430f890dc62fb9995c66fb2be8602b51e

(Sorry for the massive commit, it's kind of intertwined...)

Reply:
Updated: 23497d4

(Sorry for the massive commit, it's kind of intertwined...)

Reply:
I looked over the commit and I *think* it's OK, but as you say, it's a mammoth commit with a lot of allocates and deallocates. We should do a full memory audit at some point in the future.